{"stability":{"stable since":"1.0.0"},"apiName":"std::thread::Result","apiType":"type","apiDocumentation":{"apiDescription":["A specialized Result type for threads.","Indicates the manner in which a thread exited.","A thread that completes without panicking is considered to exit successfully."],"examples":["use std::thread;\nuse std::fs;\n\nfn copy_in_thread() -> thread::Result<()> {\n    thread::spawn(move || { fs::copy(\"foo.txt\", \"bar.txt\").unwrap(); }).join()\n}\n\nfn main() {\n    match copy_in_thread() {\n        Ok(_) => println!(\"this is fine\"),\n        Err(_) => println!(\"thread panicked\"),\n    }\n}"]}}